serializer/stream/byte_buffer.go

```diff
@@ -0,0 +1,86 @@
+package stream
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/iotaledger/hive.go/ierrors"
+)
+
+// ByteBuffer is an in-memory io.WriteSeeker implementation.
+// Adapted from https://github.com/orcaman/writerseeker.
```

Comment:
Define "adapted". Depending on that you need the original license text here.

Reply:
Added the license